### ISSUE TITLE
Orgmode + Treesitter error on ORG_TOGGLE_CHECKBOX

### ISSUE THREAD

**Issue by kevinroleke** [nvim-orgmode/orgmode#609]
### Describe the bug

I have an item like 
```
* TODO [ ] Do things
  SCHEDULED: <2023-09-08 Fri>
```

If I want to mark the checkbox, I would normally do `<Leader>Space`. Now that generates this error 

```
[orgmode] ...ck/packer/start/orgmode/lua/orgmode/utils/treesitter.lua:99: attempt to index local 'node' (a nil value)
```

Occurs regardless if `org` is configured for treesitter. 

### Steps to reproduce

orgmode config
```
require('orgmode').setup_ts_grammar()
require('orgmode').setup({
  org_agenda_files = {'~/org/*'},
  org_default_notes_file = '~/org/refile.org',
  mappings = {
    note = {
        org_note_finalize = '<Leader>w',
    }
  },
})
```

treesitter config
```
require'nvim-treesitter.configs'.setup {
  ensure_installed = { "org", "vimdoc", "javascript", "typescript", "c", "lua", "rust" },
  sync_install = false,
  auto_install = true,
  highlight = {
    enable = true,
    additional_vim_regex_highlighting = {'org'},
  },
}
```

### Expected behavior

Box gets checked

### Minimal init.lua

```lua

-- Enter your minimal_init.lua here

```

### OS / Distro

Debian 11

### Neovim version/commit

0.9.1

**Comment by jgollenz** [comment]
That's not a [checkbox](https://orgmode.org/manual/Checkboxes.html), that's an empty [priority](https://orgmode.org/manual/Priorities.html). Only plain list items can have a checkbox.

**Comment by seflue** [comment]
It happens (in case of headlines) if you add the cookie afterwards. It
doesn't appear automatically in Emacs if I recall correctly.

Am 11. September 2023 17:27:29 schrieb Jonas Gollenz
***@***.***>:

**Comment by jgollenz** [comment]
Indeed, that's a valid use-case. According to [this](https://orgmode.org/manual/The-Very-Busy-C_002dc-C_002dc-Key.html), toggling a checkbox and updating a statistics cookie is both done with `<C-c> <C-c>`. So we actually should just map it to `<C-Space>`.

> I would like to implement this

Are you talking about statistics cookies for headlines? If yes, see #572

**Comment by seflue** [comment]
> Are you talking about statistics cookies for headlines? If yes, see https://github.com/nvim-orgmode/orgmode/pull/572
Yes I do and I actually recall this abandoned PR.
> So we actually should just map it to <C-Space>.
You mean updating statistic cookies during "org-toggle-checkbox". I don't like hardcoded mappings (the user must be able to adjust them), but I'm confident, that you just used <C-Space> as a synonym for the underlying function, didn't you?

**Comment by jgollenz** [comment]
> You mean updating statistic cookies during `org-toggle-checkbox`

That is already happening. I meant doing it like we do for `org-meta-return`. Having both a mapping (be it `<C-Space>` or whatever) that does multiple things and allowing to override one of those multiple things sounds a bit complicated (although maybe doable?). I see these options:

1. have a dedicated mapping for updating the statistics cookie (which also means that it is adjustable by default)
2. introduce some sort of `org-meta-space` (since we already use <C-Space> for checkbox toggling) similar to `org-meta-return` that handles the many things that are handled by `<C-c> <C-c>` in emacs orgmode. Of course, this would not enable users to override a specific functionality, only the mapping of `org-meta-space`.
3. we add both a dedicated mapping and `org-meta-space`. We do what we do in 2., but before actually dispatching to (in this case) the logic that updates the cookie, we check if users have overridden the mapping to something else than the default of `<C-Space>`. But from the top of my head I'm quite unsure if this is even possible. If we have a mapping for `org-meta-space` and `org-toggle-checkbox` that both map to `<C-Space>`, I would expect things to fall apart.

From the docs it seems that emacs orgmode offers both the `<C-c> <C-c>` that, based on the context, does a cookie update and [a dedicated mapping](https://orgmode.org/manual/Checkboxes.html#index-C_002dc-_0023)

**Comment by jgollenz** [comment]
> Yes I do and I actually recall this abandoned PR.

Before you start, please drop me a message. I started working on this myself a while ago and I'd like to discuss some things first :+1: thanks for you contributions :slightly_smiling_face:

**Comment by seflue** [comment]
@jgollenz I think option 2 seems to be viable while not being too complicated. I'm refering to `org-toggle-checkbox` because this is the action I would use for updating the statistic cookie manually and which would make the most sense.
What currently also isn't implemented is the statistic cookie for headlines in general. For checkboxes it seems to be working quite fine - although the manual update option would be convenient if you add it as an afterthought, so you don't have to artifical toggle a checkbox in the sublist twice to update it.

Do you have an entry-point to your implementation attempt - some branch on your fork or something?

**Comment by jgollenz** [comment]
> option 2 seems to be viable

I started implementing it :+1: we can add a dedicated mapping for `org-update-statistics-cookies` later.

> entry-point to your implementation attempt

Yes, somewhere :grin: Are you in the nvim-orgmode matrix channel by any chance?

**Comment by seflue** [comment]
> Yes, somewhere 😁 Are you in the nvim-orgmode matrix channel by any chance?
No, I wasn't aware of that. Actually I have to admit, that I haven't used Matrix in the past, but if you can help me joining the channel I would be happy to do so.

**Comment by jgollenz** [comment]
> can help me joining the channel

Oh, apparently the badge is broken atm: 

![image](https://github.com/nvim-orgmode/orgmode/assets/12050098/41bc19a2-ef66-40ae-b182-548e67242db3)

Will fix that asap. Anyway, clicking on `Chat` on the [main page](https://github.com/nvim-orgmode/orgmode#nvim-orgmode) (or follow the link in the [README.md](https://github.com/nvim-orgmode/orgmode/blob/master/README.md?plain=1#L8)) should take you there. You will need a matrix account. Let me know if it's not working for you :slightly_smiling_face:
